File: halcyon-web/src/main/resources/static/js/controller/hr/hrPositionController.js
//控制层
app.controller("hrPositionController", function ($scope, $controller, hrPositionService, hrUnitService) {
    //继承
    $controller("baseController", {$scope: $scope});

    //初始化
    $scope.initForAdd = function () {
        $scope.hrPosition = {};
        $scope.getParentPositionOptions();
    }
    $scope.initForUpdate = function (id) {
        hrPositionService.queryById(id).success(function (data) {
            $scope.hrPosition = data;
        });
        $scope.getParentPositionOptions();
    }

    //crud
    //查询,初始化查询条件为空
    $scope.condition = {};
    $scope.query = function (pageNum, pageSize) {
        hrPositionService.query(pageNum, pageSize, $scope.condition).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //保存
    $scope.save = function () {
        hrPositionService.save($scope.hrPosition).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //删除
    $scope.delete = function () {
        $scope.deleteConfirm(deleteRows);
    }

    function deleteRows() {
        hrPositionService.delete($scope.selectedIds).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }

    //获取下拉框数据
    //防止ajax异步导致为空，声明一个空的
    $scope.parentPositionOptions = {data: []};
    $scope.getParentPositionOptions = function () {
        hrPositionService.getParentPositionOptions().success(function (data) {
            $scope.parentPositionOptions = {data: data};
        });
    }

    $scope.unitOptions = {data: []};
    $scope.getUnitPositions = function () {
        hrUnitService.getParentUnitOptions().success(function (data) {
            $scope.unitOptions = {data: data};
        });
    }
});

File: halcyon-base/src/main/java/com/marshal/halcyon/base/hr/controller/HrPositionController.java
package com.marshal.halcyon.base.hr.controller;

import com.marshal.halcyon.core.entity.ResponseData;
import com.marshal.halcyon.core.controller.BaseController;
import com.marshal.halcyon.base.hr.entity.HrPosition;
import com.marshal.halcyon.base.hr.service.HrPositionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * @auth: Marshal
 * @date: 2019/1/9
 * @desc:
 */
@RestController
@RequestMapping("/hr/position")
public class HrPositionController extends BaseController {

    @Autowired
    HrPositionService hrPositionService;

    @RequestMapping("/query")
    public ResponseData query(@RequestBody HrPosition condition, int pageNum, int pageSize) {
        List<HrPosition> list = hrPositionService.select(condition, pageNum, pageSize);
        return new ResponseData(list);
    }

    @RequestMapping("/submit")
    public ResponseData submit(@RequestBody HrPosition hrOrgPosition) {
        if (!getValidator().isValid(hrOrgPosition)) {
            return new ResponseData(false, getValidator().getErrors(hrOrgPosition));
        }
        hrPositionService.save(hrOrgPosition);
        return new ResponseData(true, "保存成功");
    }

    @RequestMapping("/remove")
    public ResponseData remove(@RequestParam("selectedIds") Long[] selectedIds) {
        hrPositionService.batchDelete(selectedIds);
        return new ResponseData(true, "删除成功");
    }

    @RequestMapping("/selectByPositionId")
    public HrPosition selectByPositionId(@RequestParam Long positionId) {
        return hrPositionService.selectByPrimaryKey(positionId);
    }

    @RequestMapping("/getParentPositionOptions")
    public List<Map> getParentPositionOptions() {
        return hrPositionService.getParentPositionOptions();
    }

    @RequestMapping("/selectByUnitId")
    public List<Map> selectByUnitId(@RequestParam Long unitId) {
        return hrPositionService.selectByUnitId(unitId);
    }
}
